packages/fxa-payments-server/src/components/Portal.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { memo, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+
+import './Portal.scss';
+
+type PortalProps = {
+  id: string,
+  children: React.ReactNode
+};
+
+const Portal = ({
+  id,
+  children
+}: PortalProps): React.ReactPortal | null => {
+  let el = document.getElementById(id);
+  if (! el) {
+    el = document.createElement('div');
+    el.setAttribute('class', 'portal');
+    el.setAttribute('id', id);
+    document.body.appendChild(el);
+  }
+
+  useEffect(() => {
```

Comment:
I don't know how I feel about these function react patterns... I assume `useEffect()` here registers something based on currently executing component? This all seems very opaque compared to traditional component classes. But eh...

Reply:
[useEffect](https://reactjs.org/docs/hooks-reference.html#useeffect) basically means this: After rendering the component, run the function in the 1st arg once and then whenever the set of data listed in the 2nd arg changes. The 1st arg function can return a function, which gets run as clean-up whenever the component disappears from the rendered tree.

It's kind of a replacement for a laundry list of class-based lifecycle methods like componentDidMount, componentDidUpdate, componentWillUnmount, etc.